src/app/(proper_react)/(redesign)/(public)/unsubscribe-from-monthly-report/UnsubscribeMonthlyReportView.tsx

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { useState } from "react";
+import { Button } from "../../../../components/client/Button";
+import styles from "./UnsubscribeMonthlyReport.module.scss";
+import UnsubscriptionImage from "./images/confirm-unsubscribe.svg";
+import Image from "next/image";
+import { useL10n } from "../../../../hooks/l10n";
+import { toast } from "react-toastify";
+
+export const UnsubscribeMonthlyReportView = ({ token }: { token: string }) => {
+  const [unsubscribeSuccess, setUnsubscribeSuccess] = useState(false);
+
+  const l10n = useL10n();
+  const copy = {
+    confirmation: {
+      header: l10n.getString("unsubscribe-from-monthly-report-header"),
+      body: l10n.getString("unsubscribe-from-monthly-report-body"),
+    },
+    success: {
+      header: l10n.getString("unsubscribe-success-from-monthly-report-header"),
+      body: l10n.getString("unsubscribe-success-from-monthly-report-body"),
+    },
+  };
+
+  const { header, body } = unsubscribeSuccess
+    ? copy.success
+    : copy.confirmation;
+
+  const handleUnsubscription = async () => {
+    const response = await fetch(`/api/unsubscribe-email?token=${token}`, {
+      method: "GET",
+    });
+
+    if (!response.ok) {
```

Comment:
Oh, one other thing: maybe we should also wrap the `fetch` in a `try...catch`, in case e.g. the user's internet connection is down or something else goes wrong.

(And if you want to avoid having to repeat the error code, you can then `throw` an error if `!response.ok`,)